src/nni_manager/training_service/local/localTrainingServiceForGPU.ts

```diff
@@ -91,10 +91,12 @@ class LocalTrainingServiceForGPU extends LocalTrainingService {
         trialJobDetail: TrialJobDetail,
         resource: { gpuIndices: number[] }): { key: string; value: string }[] {
         const variables: { key: string; value: string }[] = super.getEnvironmentVariables(trialJobDetail, resource);
-        variables.push({
-            key: 'CUDA_VISIBLE_DEVICES',
-            value: resource.gpuIndices.join(',')
-        });
+        if (this.gpuScheduler !== undefined) {
```

Comment:
it might be better to set CUDA_VISIBLE_DEVICES to be "" when required gpu number is 0.

Reply:
@yds05 @chicm-ms

Reply:
Yes, I agree. 
@leckie-chn , how about change to 

variables.push({
           key: 'CUDA_VISIBLE_DEVICES',
          value: this.gpuScheduler === undefined ? '' : resource.gpuIndices.join(',')
            });

Otherwise, All the gpu devices will be visible to the training process even though gpuNum is set to 0.

Reply:
@yds05 code added, see blow.

Reply:
line 94 can be removed now